### PR TITLE
Prac 5 code review request

### PR DESCRIPTION
**Pull request by chenhaoxiang-jcu**
Please kindly have a look. @hiayethta

### CODE REVIEW

File: prac_05/hex_colours.py
Comment (hiayethta) [inline prac_05/hex_colours.py:1]: Hi! Don't forget to insert a docstring at the beginning of your file w/ a brief description of your program!
I.E., Subject code, week #, purpose of program.

File: prac_05/kivy_demo.py
Comment (hiayethta) [inline prac_05/kivy_demo.py:1]: Great job w/ remembering to write docstrings for each class and function!! 🕺

File: prac_05/emails.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+"""
+
+
+def main():
+    """Get user's email address and name to print."""
+    email_to_name = {}
+    email = input('Email: ')
+    while email != '':
+        name = extract_name(email)
+        name_check = input(f"Is your name {name}? (Y/n) ").upper()
```
Comment (hiayethta) [inline prac_05/emails.py:14]: Great code! Not much for me to comment on to be honest. Perhaps capitalise the "n" in "(Y/n)" for consistency?

File: prac_05/state_names.py
Comment (hiayethta) [inline prac_05/state_names.py:1]: +1 for remembering intro docstring!

File: prac_05/state_names.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+while state_code != "":
+    try:
+        print(state_code, "is", CODE_TO_NAME[state_code])
+        state_codes.append(state_code)
+    except KeyError:
+        print("Invalid short state")
+    state_code = input("Enter short state: ").upper()
+
+abbreviation_width = max(len(state_code) for state_code in state_codes)
+[print(f"{state_code:{abbreviation_width}} is {CODE_TO_NAME[state_code]}") for state_code in state_codes]
```
Comment (hiayethta) [inline prac_05/state_names.py:22]: Maybe add a comment for what this line does? It is a bit long, so a comment would help readers understand what's going on here 🐱

**Review by hiayethta — COMMENTED** [review]
Just some minor things that could be addressed. But you did a really great job with this weeks practical tasks!

**Comment by chenhaoxiang-jcu** [comment]
Thanks for reviewing.😄 @hiayethta

File: prac_05/emails.py (hunk 1 above)
Comment (lindsaymarkward) [reply prac_05/emails.py:14]: There's a valid reason why it's Y/n on purpose :)

Comment (chenhaoxiang-jcu) [reply prac_05/emails.py:14]: > There's a valid reason why it's Y/n on purpose :)

Hi, sir. I'm curious about the reason.

Comment (lindsaymarkward) [reply prac_05/emails.py:14]: The capital letter indicates the default response. If the user just presses Enter with no other input, it is the equivalent of answering "Y". 
Answering "Y" is also OK, but you can just press Enter.
To answer "n", you need to input "n" or "N" explicitly.

Comment (chenhaoxiang-jcu) [reply prac_05/emails.py:14]: > The capital letter indicates the default response. If the user just presses Enter with no other input, it is the equivalent of answering "Y". Answering "Y" is also OK, but you can just press Enter. To answer "n", you need to input "n" or "N" explicitly.

Get it! Thank you so much, sir!

Comment (lindsaymarkward) [reply prac_05/emails.py:14]: Here's an example from a Linux system upgrade (`apt`) prompt: 
```
...
After this operation, 120 kB of additional disk space will be used.
Do you want to continue? [Y/n]
```

Comment (chenhaoxiang-jcu) [reply prac_05/emails.py:14]: I appreciate it, sir!

**Pull request merged**